src/include/iomgr/io_device.hpp

```diff
@@ -70,6 +95,31 @@ class IODevice {
     void clear();
     DriveInterface* drive_interface();
 
+    void observe_metrics(drive_iocb* iocb) {
+        if (!m_metrics.get()) { return; }
+        auto dur = get_elapsed_time_us(iocb->op_start_time);
+        switch (iocb->op_type) {
+        case DriveOpType::WRITE:
+            HISTOGRAM_OBSERVE(*m_metrics, write_lat, dur);
+            HISTOGRAM_OBSERVE(*m_metrics, write_size, iocb->size);
+            LOGINFO("write, size {}, lat {}", iocb->size, dur);
```

Comment:
Change to debug or trace log after you've finished debugging? Otherwise it would be too much log, right.

Reply:
yes this is more for debugging ATM. will lower them to trace later